drivers/adc/ad4858/ad4858.h

```diff
@@ -0,0 +1,407 @@
+#define AD4858_REG_DEVICE_CTRL  			(AD4858_R1B | 0x25)
+#define AD4858_REG_PACKET	 				(AD4858_R1B | 0x26)
+#define AD4858_REG_OVERSAMPLE  				(AD4858_R1B | 0x27)
+#define AD4858_REG_SEAMLESS_HDR  			(AD4858_R1B | 0x28)
+#define AD4858_REG_CH_SLEEP  				(AD4858_R1B | 0x29)
+#define AD4858_REG_CH_SOFTSPAN(chn)  		(AD4858_R1B | (0x2A + (0x12 * chn)))
+#define AD4858_REG_CH_OFFSET(chn)  			(AD4858_R3B | (0x2B + (0x12 * chn)))
+#define AD4858_REG_CH_GAIN(chn)  			(AD4858_R2B | (0x2E + (0x12 * chn)))
+#define AD4858_REG_CH_PHASE(chn)  			(AD4858_R2B | (0x30 + (0x12 * chn)))
+#define AD4858_REG_CH_OR(chn)  				(AD4858_R3B | (0x32 + (0x12 * chn)))
+#define AD4858_REG_CH_UR(chn)  				(AD4858_R3B | (0x35 + (0x12 * chn)))
+#define AD4858_REG_CH_TESTPAT(chn)  		(AD4858_R4B | (0x38 + (0x12 * chn)))
+
+/** AD4858_REG_INTERFACE_CONFIG_A bit masks */
+#define AD4858_SW_RESET_MSK			NO_OS_BIT(7) | NO_OS_BIT(0)
+#define AD4858_SDO_ENABLE_MSK		NO_OS_BIT(4)
+#define AD4858_ADDR_ASCENSION_MSK	NO_OS_BIT(5)
+
+/** AD4858_REG_INTERFACE_CONFIG_B bit masks */
+#define AD4858_SINGLE_INST_MSK		NO_OS_BIT(7)
+
+/** AD4858_REG_DEVICE_CONFIG bit masks */
+#define AD4858_OPERATING_MODES_MSK	NO_OS_GENMASK(1,0)
+#define AD4858_STATUS_BIT0_MSK		NO_OS_BIT(4)
+#define AD4858_STATUS_BIT1_MSK		NO_OS_BIT(5)
+#define AD4858_STATUS_BIT2_MSK		NO_OS_BIT(6)
+#define AD4858_STATUS_BIT3_MSK		NO_OS_BIT(7)
+
+/** AD4858_REG_TRANSFER_CONFIG bit masks */
+#define AD4858_KEEP_STRM_LEN_MSK	NO_OS_BIT(2)
+
+/** AD4858_REG_INTERFACE_CONFIG_C bit masks */
+#define AD4858_ACTIVE_INF_MODE_MSK	NO_OS_GENMASK(3,2)
+#define AD4858_CRC_ENABLE_MSK		NO_OS_GENMASK(7,6)
+
+/** AD4858_REG_INTERFACE_STATUS_A bit masks */
+#define AD4858_ADDR_INVALID_ERR_MSK		NO_OS_BIT(0)
+#define AD4858_WR_TO_RD_ONLY_ERR_MSK	NO_OS_BIT(2)
+#define AD4858_CRC_ERR_MSK				NO_OS_BIT(3)
+#define AD4858_CLK_COUNT_ERR_MSK		NO_OS_BIT(4)
+#define AD4858_NOT_READY_ERR_MSK		NO_OS_BIT(7)
+
+/** AD4858_REG_SPI_CONFIG_D bit masks */
+#define AD4858_CSDO_ON_SDO_MSK		NO_OS_BIT(0)
+
+/** AD4858_REG_PACKET bit masks */
+#define AD4858_TEST_PATTERN_MSK		NO_OS_BIT(2)
+#define AD4858_PACKET_FORMAT_MSK	NO_OS_GENMASK(1,0)
+
+/** AD4858_REG_OVERSAMPLE bit masks */
+#define AD4858_OS_ENABLE_MSK		NO_OS_BIT(7)
+#define AD4858_OS_RATIO_MSK			NO_OS_GENMASK(3,0)
+
+/** AD4858_REG_CH_SOFTSPAN bit masks */
+#define AD4858_SOFTSPAN_MSK			NO_OS_GENMASK(3,0)
+
+/** Miscellaneous Definitions */
+#define AD4858_REG_RD_BIT_MSK		NO_OS_BIT(7)
+#define AD4858_PRODUCT_ID_L			0x60
+#define AD4858_PRODUCT_ID_H			0x00
+#define AD4858_NUM_CHANNELS			8
+#define AD4858_DEF_CHN_SOFTSPAN		0xf
+#define AD4858_DEF_CHN_OFFSET		0x0
+#define AD4858_DEF_CHN_GAIN			0x8000
+#define AD4858_DEF_CHN_PHASE		0x0
+#define AD4858_DEF_CHN_OR			0x7ffff0
+#define AD4858_DEF_CHN_UR			0x800000
+
+/**
+ * @enum ad4858_operating_mode
+ * @brief Operating modes
+ */
+enum ad4858_operating_mode {
+	AD4858_NORMAL_OP_MODE,
+	AD4858_LOW_POWER_OP_MODE=0x3,
```

Comment:
if one member of the enum is explicitly initialized, then all of them should be.